src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1122,27 +1191,27 @@ _Success_(return ) bool OutputStateMachineEngine::_GetDeviceStatusOperation(_In_
 }
 
 // Routine Description:
-// - Retrieves the listed private mode params be set/reset by DECSET/DECRST
+// - Converts the untyped array of numeric parameters into an array of the specified type.
 // Arguments:
-// - rPrivateModeParams - Pointer to array space (expected 16 max, the max number of params this can generate) that will be filled with valid params from the PrivateModeParams enum
-// - pcParams - Pointer to the length of rPrivateModeParams on the way in, and the count of the array used on the way out.
+// - rgTypedParams - Pointer to array space (expected 16 max, the max number of params this can generate) that will be filled with valid params from the PrivateModeParams enum
+// - pcParams - Pointer to the length of rgTypedParams on the way in, and the count of the array used on the way out.
 // Return Value:
-// - True if we successfully retrieved an array of private mode params from the parameters we've stored. False otherwise.
-_Success_(return ) bool OutputStateMachineEngine::_GetPrivateModeParams(_In_reads_(cParams) const unsigned short* const rgusParams,
-                                                                        const unsigned short cParams,
-                                                                        _Out_writes_(*pcParams) DispatchTypes::PrivateModeParams* const rgPrivateModeParams,
-                                                                        _Inout_ size_t* const pcParams) const
+// - True if we successfully retrieved an array of strongly-typed params from the parameters we've stored. False otherwise.
+template<typename TParamType>
+_Success_(return ) bool OutputStateMachineEngine::_GetTypedParams(_In_reads_(cParams) const unsigned short* const rgusParams,
+                                                                  const unsigned short cParams,
+                                                                  _Out_writes_(*pcParams) TParamType* const rgTypedParams,
+                                                                  _Inout_ size_t* const pcParams) const
 {
     bool fSuccess = false;
-    // Can't just set nothing at all
     if (cParams > 0)
     {
         if (*pcParams >= cParams)
         {
             for (size_t i = 0; i < cParams; i++)
             {
-                // No memcpy. The parameters are shorts. The graphics options are unsigned ints.
-                rgPrivateModeParams[i] = (DispatchTypes::PrivateModeParams)rgusParams[i];
+                // No memcpy because the parameters are shorts, and the destination type may be a different size.
+                rgTypedParams[i] = static_cast<TParamType>(rgusParams[i]);

```

Comment:
should narrowing into a shorter `TParamType` be worth noting? We may want to `gsl::narrow_cast` to be safe, or consider its throwing variant if we expect that to not be a valid case